internal/rbd/rbd_util.go

```diff
@@ -2071,3 +2071,27 @@ func strategicActionOnLogFile(ctx context.Context, logStrategy, logFile string)
 		log.ErrorLog(ctx, "unknown cephLogStrategy option %q: hint: 'remove'|'compress'|'preserve'", logStrategy)
 	}
 }
+
+// genVolFromVolIDWithMigration populate a rbdVol structure based on the volID format.
+func genVolFromVolIDWithMigration(
```

Comment:
function name says this is for migration but you are generating rbdVol for both cases.

Reply:
it is same model as what @nixpanic suggested for credential support , similar one to @newUserCredentialsWithMigration to make it clean.Otherwise a if ...else in the caller does the job.